packages/react-card/Spec.md

```diff
@@ -0,0 +1,482 @@
+| ----------- | ------------------------------------------------------------------------ |
+| Card        | Wrapper for the main content of a card.                                  |
+| CardHeader  | Optional header slot for the card.                                       |
+| CardBody    | Main slot of the Card                                                    |
+| CardPreview | Previewing the document/article presented in the card through an image   |
+| CardFooter  | Optional footer slot for the card. (non-applicable for horizontal cards) |
+
+### Card
+
+#### Anatomy
+
+![visual anatomy of the Card component](./assets/Card.png)
+
+#### API
+
+| Property    | Values                                                                               | Default    | Purpose                                                                                   |
+| ----------- | ------------------------------------------------------------------------------------ | ---------- | ----------------------------------------------------------------------------------------- |
+| orientation | `vertical`, `horizontal`                                                             | `vertical` | Orientation of the card                                                                   |
+| size        | `smallest`, `smaller`, `small`, `medium`, `large`                                    | `medium`   | Define the minimum size of the card. Smaller sizes only apply to horizontal card          |
+| scale       | `fixed`, `auto-width`, `auto-height`, `auto`, `fluid-width`, `fluid-height`, `fluid` | `auto`     | Manages how the card handles it's scaling depending on the content                        |
+| onClick     | function                                                                             | undefined  | Function to call when clicking the card. Makes the card assume the interactive appearance |
+| appearance  | `filled`, `filled-alternative`, `outline`, `subtle`                                  | `filled`   | Define the appearance of the card                                                         |
+| selectable  | boolean                                                                              | false      | Makes the card selectable by adding a checkbox to the _Actions_ area                      |
+| selected    | boolean                                                                              | false      | Set to `true` if card is selected                                                         |
+| expandable  | boolean                                                                              | false      | Allow card to expand to show whole content                                                |
+| contextMenu | _slot_                                                                               | undefined  | Slot for the context menu, when right-clicking the card                                   |
```

Comment:
I do not think that a contextMenu should be in a `Card` as a slot. The current `Menu` implementation encourages composition of components through JSX. What is wrong with this kind of API to support a context menu scenario for card ? 

https://codesandbox.io/s/optimistic-cache-975ji?file=/src/App.js

Reply:
Here is another example if we don't want embed the `Card` inside a `MenuTrigger` for a context menu: https://codesandbox.io/s/falling-tree-oh6o0?file=/src/App.js

In order to support a context menu slot we would also need to bundle `react-menu` somehow with it adding to bloat.

I would say that we should look for some use case that we cannot achieve with the existing APIs that we have and then see what is the best way to achieve them. Menu slots in v0/8 are generally very complicated an hard to memoize/virtualize since they are just js objects. The 2 methods I've proposed here all leverage plain JSX.

Reply:
Yeah this makes much more sense and option 2 also allows us to reuse the menu for when you click an ellipsis menu scenario. Thanks for the example @ling1726, do you think this should be covered in the spec or docs (or both)?

Reply:
This should be covered in the docs IMO.

Reply:
Awesome, then I'll just add a section to explain this should be covered with Menu instead and then in the future use your examples in the docs. Thanks!